### PR TITLE
fix: Use material towards target in CKF

### PR DESCRIPTION
**Pull request by andiwand**
reopened https://github.com/acts-project/acts/pull/2318 after https://github.com/acts-project/acts/pull/2321

The CKF did not pick up the material in-between the first surface and the target surface. This PR is trying to fix that.

Note: I had to use a not so nice hack and call the navigation pre step routine before we step towards the target. I don't see an easy way around that.

blocked by
- https://github.com/acts-project/acts/pull/2336

### CODE REVIEW

**Comment by andiwand** [comment]
> This fails with the navigation error that you've been looking into, right @andiwand? Does this change outputs (probably)?

yeah exactly this fails in Athena and I am working on a fix. This does change outputs but not dramatically. Highly depends on the seeder in use which might place you further inside the detector

**Comment by paulgessinger** [comment]
Sorry, misclick on the close button 😅.

It does change both the hashes and the physmon outputs.

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp [hunk 1]
```diff
@@ -353,12 +353,12 @@ class CombinatorialKalmanFilter {
     /// @param stepper is the stepper in use
     /// @param navigator is the navigator in use
     /// @param result is the mutable result state object
-    /// @param logger the logger object to be used
+    /// @param _logger the logger object to be used
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:356]: Meh, formatting. Don't apply this 1:1.

**Comment by andiwand** [comment]
will update the reference after ci is don

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691892) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691891) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691890) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691889) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691888) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691887) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691886) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691885) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691884) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691883) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691882) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691881) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691880) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691879) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691878) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691877) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32691876) |

**Comment by andiwand** [comment]
athena passed :scream: @paulgessinger do you know if there are some more athena internal tests that could fail after the nightly?

**Comment by paulgessinger** [comment]
@andiwand Possible, but not likely I would say. We run most (if not all?) of what the Athena CI runs, and the workflow tests should sort of be indicative of what ART ends up looking at.

**Comment by timadye** [comment]
This has now been merged into Athena `main`, discussed in [ATLIDTRKCP-500](https://its.cern.ch/jira/browse/ATLIDTRKCP-500) (I hope it's OK to refer to ATLAS JIRA tickets here, even if these may not be accessible to all).

We see an increase in the track errors (all except q/pT), but not in the resolutions. This leads to improved (closer to 1) pulls, so it looks like an improvement overall. Is this what you saw from physmon before the references were updated?

**Comment by andiwand** [comment]
great! with track errors you mean the covariance got a bit bigger and the resolution is unaffected means that the fit still behaves the same, right? I also observed that the pulls are better behaved afterwards

@timadye

**Comment by timadye** [comment]
@andiwand, yes it was just changes in the covariance, so better pulls.

**Pull request merged**